server/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/TextPreprocessorImpl.java

```diff
@@ -101,14 +100,14 @@ public TextPreprocessorImpl(
    */
   @NonNull
   @Override
-  @CheckThreadInterruption
   public ResultWithErrors<ExtendedDocument> process(
       @NonNull String documentUri,
       @NonNull String cobolCode,
       @NonNull Deque<CopybookUsage> copybookStack,
       @NonNull CopybookProcessingMode copybookProcessingMode,
       @NonNull Deque<List<Pair<String, String>>> recursiveReplaceStmtStack,
       @NonNull List<Pair<String, String>> replacingClauses) {
+    ThreadInterruptionUtil.checkThreadInterrupted("TextPreprocessorImpl#process");
```

Comment:
You may  do import static, then just calling `checkThreadInterrupted(methodName);`  in  all places/classes

Reply:
It won't work for `@UtilityClass` and other annotations that produce static methods.

Reply:
works fine

Reply:
I tried this. The IDE doesn't have an issue but it fails while building the project. It throws "cannot find symbol" at static import.